src/compression.rs

```diff
@@ -8,13 +8,13 @@
 
 // Part of the file is borrowed from <https://github.com/seanmonstar/warp/pull/513>*
 
-#[cfg(feature = "compression-brotli")]
+#[cfg(any(feature = "compression", feature = "compression-brotli"))]
```

Comment:
Nit: With the `compression` feature activating the `compression-brotli` feature, I’m fairly certain that this change (and a bunch of similar ones) is unnecessary. Unless you are doing this for readability purposes.

Reply:
Correct, I'm doing this just for readability because using `compression` already implies the other ones.

But it could be removed without a problem.

Feel free to clean this up if you think so.